### ISSUE TITLE
Warning:  Ambiguous class resolution

### ISSUE THREAD

**Issue by enduro44** [laravel-json-api/laravel#218]
Hi,
I'm using `laravel-json-api` in my project and everything works just fine. 
Now I'm trying to use some other package that has dependency of `neomerx/json-api` and I run into this problem. When doing composer update I get several messages like this:
```
Warning:  Ambiguous class resolution, "Neomerx\JsonApi\Contracts\Encoder\EncoderInterface" was found in both "/vendor/laravel-json-api/neomerx-json-api/src\Contracts\Encoder\EncoderInterface.php" and "/vendor/neomerx/json-api/src\Contracts\Encoder\EncoderInterface.php", the first will be used.
```
and 
```
PHP Fatal error:  Cannot redeclare Neomerx\JsonApi\I18n\format() (previously declared in\web\vendor\laravel-json-api\neomerx-json-api\src\I18n\format.php:31) in \web\vendor\neomerx\json-api\src\I18n\form
at.php on line 31
```

Is it possible to use `laravel-json-api` together with any other package that uses `neomerx/json-api`?
What would be the best solution for this problem?

**Comment by lindyhopchris** [comment]
This is because we had to fork the `neomerx/json-api` package. So you're getting `laravel-json-api/neomerx-json-api` installed for our package, and the other dependency will be installing `neomerx/json-api`. So you'll end up with two neomerx packages installed, which then conflict with each other.

There's nothing I can do about this unfortunately. The `neomerx/json-api` package isn't maintained, and there was a bug in it that needed to be fixed for our use of it. Which is why I forked it and fixed it. 

The solution is to get the other package that relies on it to use our fork. At the moment by relying on `neomerx/json-api` they are relying on an unmaintained package.
